middleware.js

```diff
@@ -198,12 +197,14 @@ module.exports = function(options) {
 
       mkdirp(dirname(cssPath), '0700', function(err) {
         if (err) {
-          return error(err);
+          error(err);
+          cssDone = true;
+          return doneWriting();
```

Comment:
The problem is that nothing calls `next()` and nothing touches `req` after this point, so the request is left to time out.

Even if someone supplies `options.error` and we call them, they don't have access to `req` or `next`, so they can't do anything with the request.

In most other places, the call to `error()` is followed by `next()`, but not here; I can't imagine right now what dependency on this behaviour someone might have taken – can you please illustrate what you mean?

Propagating to the next middleware in case of error is the expected behaviour – we are a middleware after all.